packages/tests/server/react/rsc-prefetch.test.tsx

```diff
@@ -78,36 +87,48 @@ const ctx = konn()
   .done();
 
 test('rsc prefetch helpers', async () => {
-  const { client, App, trpc, HydrateClient } = ctx;
+  const { client, App, trpc, HydrateClient, getQueryClient } = ctx;
 
   const fetchSpy = vi.spyOn(globalThis, 'fetch');
 
   function MyComponent() {
-    const { data } = client.post.byId.useQuery({
+    const q1 = client.post.byId.useQuery({
       id: '1',
     });
 
-    return <>{data}</>;
+    const q2 = client.post.byId.useQuery({
+      id: '2',
+    });
+
+    return (
+      <>
+        {q1.data}
+        {q2.data}
+      </>
+    );
   }
 
   // Imaginary RSC prefetch parent component
   function Parent() {
     void trpc.post.byId.prefetch({ id: '1' });
+    void trpc.post.byId.prefetch({ id: '2' });
     return (
       <HydrateClient>
         <MyComponent />
       </HydrateClient>
     );
   }
 
-  const utils = render(
+  const utils1 = render(
     <App>
       <Parent />
     </App>,
   );
   await waitFor(() => {
-    expect(utils.container).toHaveTextContent(`__result`);
+    expect(utils1.container).toHaveTextContent('__result1');
+    expect(utils1.container).toHaveTextContent('__result2');
   });
+  expect(ctx.postByIdInvokations).toHaveBeenCalledTimes(2);
 
   // Should not have fetched from CC but taken promise from server client
   expect(fetchSpy).toHaveBeenCalledTimes(0);
```

Comment:
Can you verify this is the case "in the real world" by running the example and checking that it doesn't fetch?

I should add an e2e test 😅